datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -529,7 +529,12 @@ impl GroupedHashAggregateStream {
             })
             .collect();
 
-        let name = format!("GroupedHashAggregateStream[{partition}]");
+        let agg_fn_names = aggregate_exprs
+            .iter()
+            .map(|expr| expr.name().to_string())
+            .collect::<Vec<String>>()
```

Comment:
minor: I think you should not need to allocate a `String` explicitly, you can join on slices
```suggestion
            .map(|expr| expr.name())
            .collect::<Vec<_>>()
```